### ISSUE TITLE
Wrong parse result from single scalar document

### ISSUE THREAD

**Issue by fktn-k** [fktn-k/fkYAML#406]
### Description

Parsing a YAML document which contains only one scalar produces a mapping with the scalar as a key to a null scalar, but the parse result must be a scalar.

### Reproduction steps

Run the following code:  

```cpp
char input[] = “foo“;
auto n = fkyaml::node::deserialize(input);
auto scalar = n.get_value<std::string>(); // This line throws a fkyaml::type_error since n is a mapping.
```

### Expected vs. actual results

The parse result must be a scalar, but is a mapping.

### Minimal code example

```Shell
See the reproduction steps.
```

### Error messages

```Shell
type_error: The node value is not a string. type=mapping
```

### Compiler and operating system

GCC 11.4.0 on Ubuntu 22.04 LTS

### Library version

develop HEAD

### Validation

- [X] The bug also occurs if the latest version from the [`develop`](https://github.com/fktn-k/fkYAML/tree/develop) branch is used.
- [X] I can successfully [compile and run the unit tests](https://github.com/fktn-k/fkYAML#how-to-execute-the-unit-tests).

**Comment by fktn-k** [comment]
This bug was found when trying to reproduce the issue #405.

**Comment by fktn-k** [comment]
addressed in the PR #411
